crates/torii/core/src/engine.rs

```diff
@@ -340,6 +340,10 @@ impl<P: Provider + Sync> Engine<P> {
                 self.process_block(block_number, blocks[&block_number]).await?;
                 last_block = block_number;
             }
+
+            if self.db.query_queue.queue.len() >= 1000 {
```

Comment:
We should make this as a constant or some sort of variable exposed on the torii cli

Reply:
i think its an implementation detail users most likely wont change it so no need to expose on CLI, but agree on making it a constant

Reply:
The performance on that would likely change based on the underlying db. We may for now have a const that is well calibrated to `SQLite`, and expose it later.